src/Avalonia.Controls/Primitives/ScrollBar.cs

```diff
@@ -506,5 +513,211 @@ private void UpdatePseudoClasses(Orientation o)
+            contextMenu.Items.Add(new Separator());
+            contextMenu.Items.Add(CreateMenuItem("StringScrollBarRightEdge", "RightEdge", () => ScrollToRightEndAction(this)));
+            contextMenu.Items.Add(CreateMenuItem("StringScrollBarLeftEdge", "LeftEdge", () => ScrollToLeftEndAction(this)));
+            contextMenu.Items.Add(new Separator());
+            contextMenu.Items.Add(CreateMenuItem("StringScrollBarPageRight", "PageRight", () => PageRightAction(this)));
+            contextMenu.Items.Add(CreateMenuItem("StringScrollBarPageLeft", "PageLeft", () => PageLeftAction(this)));
+            contextMenu.Items.Add(new Separator());
+            contextMenu.Items.Add(CreateMenuItem("StringScrollBarScrollRight", "ScrollRight", () => LineRightAction(this)));
+            contextMenu.Items.Add(CreateMenuItem("StringScrollBarScrollLeft", "ScrollLeft", () => LineLeftAction(this)));
+            return contextMenu;
+        }
+
+        private MenuItem CreateMenuItem(string resourceKey, string automationId, Action action)
+        {
+            var menuItem = new MenuItem
+            {
+                Command = new SimpleCommand(action)
+            };
+            
+            if (this.TryFindResource(resourceKey, out var resource) && resource is string header)
+            {
+                menuItem.Header = header;
+            }
+            else
+            {
+                // If we can't get one, fall back to the resource key itself
+                // So at least something is there.
+                menuItem.Header = resourceKey.Replace("StringScrollBar", "").Replace("ScrollBar", "");
+            }
+            
+            AutomationProperties.SetAutomationId(menuItem, automationId);
+            return menuItem;
+        }
+
+        private static void ScrollHereAction(ScrollBar scrollBar)
+        {
+            if (scrollBar.Track != null)
+            {
+                var track = scrollBar.Track;
+                var trackLength = scrollBar.Orientation == Orientation.Vertical ? track.Bounds.Height : track.Bounds.Width;
+                var thumbLength = scrollBar.Orientation == Orientation.Vertical ? track.Thumb?.Bounds.Height ?? 0 : track.Thumb?.Bounds.Width ?? 0;
+                var clickPosition = scrollBar.Orientation == Orientation.Vertical ? scrollBar._lastRightClickPosition.Y : scrollBar._lastRightClickPosition.X;
+                
+                if (trackLength > thumbLength)
+                {
+                    var ratio = clickPosition / trackLength;
+                    var range = scrollBar.Maximum - scrollBar.Minimum;
+                    var value = scrollBar.Minimum + (ratio * range);
+                    scrollBar.SetCurrentValue(ValueProperty, Math.Max(scrollBar.Minimum, Math.Min(scrollBar.Maximum, value)));
+                    scrollBar.OnScroll(ScrollEventType.ThumbTrack);
+                }
+            }
+        }
+
+        private static void ScrollToTopAction(ScrollBar scrollBar)
```

Comment:
Adding the suffix Action at the end doesn't really follow any conventions I'm aware of. Additionally, there is an argument to be made that these could become public in the future. and also... not static.